TM-51  Read and write test results to artifactory.

barrylapthorn opened the pull request:
👮🏻👮🏻👮🏻 !!!! DESCRIBE YOUR CHANGES HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻


# PR Checklist:

- [ ] Have you run the unit, integration and smoke tests as described [here](https://docs.corda.net/head/testing.html)?
- [ ] If you added public APIs, did you write the JavaDocs?
- [ ] If the changes are of interest to application developers, have you added them to the [changelog](https://docs.corda.net/head/changelog.html) (`/docs/source/changelog.rst`), and potentially the [release notes](https://docs.corda.net/head/release-notes.html) (`/docs/source/release-notes.rst`)?
- [ ] If you are contributing for the first time, please read the [contributor agreement](https://docs.corda.net/head/contributing.html) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://docs.corda.net/head/contributing.html#merging-the-changes-back-into-corda).

Thanks for your code, it's appreciated! :)

barrylapthorn committed:
TM-51  Prep for reading and writing test results to artifactory.

barrylapthorn committed:
TM-51  Tests from target branch if no tests for current branch

barrylapthorn committed:
TM-51  Placeholder for test averaging over runs.

barrylapthorn committed:
TM-51  Replace slashes in branch names used as tags.

barrylapthorn committed:
TM-51  More placeholder work for the mean duration work.

barrylapthorn committed:
TM-51  Write out average tests results as as csv.

The csv file should grow and be updated on each run.  This includes whether or not we are running unit tests, integration tests and so on.

barrylapthorn committed:
TM-51  Comment out old junit test archiving, add more comments.

barrylapthorn committed:
TM-51  Zip task needs to depend on a csv creation task.

If there isn't a csv file present, then the zip task doesn't run due to 'NO-SOURCE'

barrylapthorn committed:
TM-51  Zip task should ignore empty dirs

barrylapthorn committed:
TM-51  Fix up loading of test results.

We were looking for the wrong artifact name.
Add a bit more logging.

barrylapthorn committed:
TM-51  Fix up possible problem with allocating by class distribution.

If we encounter a class we haven't seen before, there won't be any tests.
This means we should give it some weight.  '1' is far too small.

barrylapthorn committed:
TM-51  Test that we are definitely increment the run count.

Tracking down whether the zipped csv file should have incremented.

roastario commented on buildSrc/src/main/groovy/net/corda/testing/KubesTest.java:
```diff
@@ -327,9 +327,19 @@ private void waitForPodToStart(KubernetesClient client, Pod pod) {
     }
 
     private String[] getBuildCommand(int numberOfPods, int podIdx) {
+        final String gitBranch = " -Dgit.branch=" +TestDurationArtifacts.getGitBranch();
```
hmmm, could we extract the "-D" into the command line? Just makes it obvious it's being passed in as a System property.

barrylapthorn committed:
TM-51  Better default value for missing test/class names.

Begin by using mean unit test duration, but we have the option to bump
that to the mean class unit tests duration.

barrylapthorn committed:
TM-51  More debug information around csv writing.

We should be incrementing the tests.

barrylapthorn committed:
TM-51  Reload the csv before updating it.

barrylapthorn committed:
TM-51  Reduce verbosity of logging.

barrylapthorn committed:
TM-51  Reinstate unit tests.  Remove logging verbosity.

barrylapthorn committed:
TM-51  Load tests from artifactory in memory and avoid interim file.

barrylapthorn committed:
TM-51  Better handling of zero duration tests.

Ensure we return zero times from junit artifacts which may either be zero or have no recorded time.  Before writing the tests duration csv file, store those with a known time, and then store those with zero using the average time.

barrylapthorn committed:
TM-51  Log whether we have recorded a test.

Tracking down the curious case where we seem to not be rerunning the
same set of tests on the second run.

barrylapthorn committed:
TM-51  Capture junit files as well.

Trying to track down whether some tests are intermittently run.

barrylapthorn committed:
TM-51  Change task dependencies to ensure ziptask is triggered.

barrylapthorn committed:
TM-51  Remove test assertion, and trigger build

barrylapthorn commented:
Changes:

* store the durations in `long` nanos, rather than a `double`
* `List<Tuple<String, Double>>` == test durations now inside a class called `Tests`
* Lots of logging statements that need removing,
* a couple of erroneous `.idea` files that I'll remove (again, again).

Outstanding:  the writing task (line 101 of DistrubtedTest.groovy), doesn't appear to have access to all junit xml files from the pods, implying that the task that gathers them from the pods doesn't always work.

barrylapthorn committed:
TM-51  Add corda/enterprise to artifactory tag name.

Moved properties to own file.

barrylapthorn committed:
TM-51  Remove unnecessary mean class-based duration.

barrylapthorn committed:
TM-51  Add more BucketingAllocator tests.

We need these to nail down its behaviour some more.

barrylapthorn committed:
TM-51  Further log information.

We don't seem to be finding the tests in the 'production' runs which is odd.

barrylapthorn committed:
TM-51  corda type double set?

barrylapthorn committed:
TM-51  do not set the project type in the properties.

SRP and all that.

barrylapthorn committed:
TM-51  better plan reporting

barrylapthorn committed:
TM-51  duration may be zero

Another runtime problem that doesn't show in tests.

barrylapthorn committed:
TM-51  better plan reporting

roastario committed:
Merge remote-tracking branch 'origin/release/os/4.3' into TM-51/store-test-results-in-artifactory

# Conflicts:
#	Jenkinsfile
#	build.gradle
#	buildSrc/src/main/groovy/net/corda/testing/KubesTest.java

roastario committed:
fix missing space after image id

roastario committed:
fix merge issue in DistributedTesting

barrylapthorn committed:
TM-51  remove unused code when GET/PUT-ting to Artifactory.

barrylapthorn committed:
TM-51  put tasks in gradle group and tidy up zip task creation

roastario committed:
Merge remote-tracking branch 'origin/release/os/4.3' into TM-51/store-test-results-in-artifactory

# Conflicts:
#	buildSrc/src/main/groovy/net/corda/testing/KubesTest.java

barrylapthorn committed:
TM-51 Fix the junit XML path.

barrylapthorn committed:
TM-51 Fix the task graph

barrylapthorn committed:
TM-51 Less logging
